### PR TITLE
Add upconversion docs

### PR DESCRIPTION
**Pull request by bartelink**
Extending example provided by @kevingentile in #64, prompted by @ameier38 in the discord

### CODE REVIEW

**Comment by bartelink** [comment]
Any and all suggestions welcomed; I hate writing docs like these, no matter how important they are :D

File: README.md [hunk 1]
```diff
@@ -566,33 +568,148 @@ There are two events that we were not able to decode, for varying reasons:
...
+As a system evolves, the types used for events will inevitably undergo changes too. There are thorough guides such as
+[Versioning in an Event Sourced System by Greg Young](https://leanpub.com/esversioning); this will only scratch the surface,
+with some key F# snippets.
+
+High level rules:
+  1. The most important rule of all is that you never want to relinquish Total Matching, i.e. never add a `_` catch all case
+to a match expression.
+  2. The simplest way to add a new field in a backward compatible manner is by adding it as an `option` and then using
+     pattern matching to handle presence or absence of the value. 
+  3. Where it becomes impossible to use the serialization-time conversion mechanisms such as [`JsonIsomorphism`](#jsonisimorphism)
```
Comment (nordfjord) [inline README.md:583]: I'm not sure how JsonIsomorphism fits into the versioning of events. Can you provide an example of it being used for that purpose?

Comment (bartelink) [reply README.md:583]: Hm, don't have one to hand - see the section https://github.com/jet/fscodec#typesafeenum-fallback-converters-using-jsonisomorphism though
in that case, it uses `string` as the proxy type, but that can equally be a record type. So you define a record to consume what's there, and provide conversions to/from the domain side type, with as much logic as you want.

Comment (bartelink) [reply README.md:583]: example: https://github.com/jet/propulsion/blob/059dc1b80cc6468ae5610f7847cfcdfdf2035cdc/src/Propulsion.DynamoStore/AppendsIndex.fs#L17

Comment (bartelink) [reply README.md:583]: Adding link...

**Pull request merged**